<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>ExitValueException.java</title>
    <link rel="stylesheet" type="text/css" href="../../css/style.css">
    <link href="https://fonts.googleapis.com/css?family=Roboto+Mono:400,700" rel="stylesheet">
</head>
<body>

<div class="container">
    <div id="code_area" class="container_0">
<div>1&emsp;&emsp;package org.unix4j.sort.command;</div>
<div>2&emsp;&emsp;</div>
<div>3&emsp;&emsp;import org.unix4j.sort.io.FileInput;</div>
<div>4&emsp;&emsp;import org.unix4j.sort.io.Input;</div>
<div>5&emsp;&emsp;</div>
<div>6&emsp;&emsp;/**</div>
<div>7&emsp;&emsp;&nbsp; * An exception thrown when a command terminates with an error; in Unix, the</div>
<div>8&emsp;&emsp;&nbsp; * command would return a value different from zero. Note that the Java VM does</div>
<div>9&emsp;&emsp;&nbsp; * NOT terminate and exit if such an exception occurs.</div>
<div>10&emsp;&emsp;&nbsp; */</div>
<div>11&emsp;&emsp;public class ExitValueException extends RuntimeException {</div>
<div>12&emsp;&emsp;</div>
<div>13&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    private static final long serialVersionUID = 2153738224794893827L;</div>
<div>14&emsp;&emsp;</div>
<div>15&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    private final int exitValue;</div>
<div>16&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    private Input input;</div>
<div>17&emsp;&emsp;</div>
<div>18&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    /**</div>
<div>19&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * Constructor with message and non-zero exit value.</div>
<div>20&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     *</div>
<div>21&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @param message   the error message</div>
<div>22&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @param exitValue the exit value, must be different from zero</div>
<div>23&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @throws IllegalArgumentException if {@code exitValue==0}</div>
<div>24&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     */</div>
<div>25&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    public ExitValueException(String message, int exitValue) {</div>
<div>26&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        super(message);</div>
<div style="background-color:limegreen;">27&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<b>        if(exitValue == 0) {&nbsp;&#8594; [MERGE, UNIQUE, CHECK]</b></div>
<div>28&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;            throw new IllegalArgumentException("exit value must be a non-zero value");</div>
<div>29&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        }</div>
<div>30&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        this.exitValue = exitValue;</div>
<div>31&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    }</div>
<div>32&emsp;&emsp;</div>
<div>33&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    /**</div>
<div>34&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * Constructor with message, a non-zero exit value and a causing exception.</div>
<div>35&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     *</div>
<div>36&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @param message   the error message</div>
<div>37&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @param exitValue the exit value, must be different from zero</div>
<div>38&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @param cause     the exception that caused the error termination of the command</div>
<div>39&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @throws IllegalArgumentException if {@code exitValue==0}</div>
<div>40&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     */</div>
<div>41&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    public ExitValueException(String message, int exitValue, Throwable cause) {</div>
<div>42&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        super(message, cause);</div>
<div>43&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        if(exitValue == 0) {</div>
<div>44&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;            throw new IllegalArgumentException("exit value must be a non-zero value");</div>
<div>45&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        }</div>
<div>46&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        this.exitValue = exitValue;</div>
<div>47&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    }</div>
<div>48&emsp;&emsp;</div>
<div>49&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    /**</div>
<div>50&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * Returns the exit value, an int value or error code different from zero</div>
<div>51&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * that was passed to the constructor of this exception.</div>
<div>52&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     *</div>
<div>53&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @return the non-zero exit or error code value</div>
<div>54&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     */</div>
<div>55&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    public int getExitValue() {</div>
<div>56&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        return exitValue;</div>
<div>57&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    }</div>
<div>58&emsp;&emsp;</div>
<div>59&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    @Override</div>
<div>60&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    public String getMessage() {</div>
<div>61&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        if(input == null) {</div>
<div>62&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;            return super.getMessage();</div>
<div>63&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        }</div>
<div>64&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        return super.getMessage() + "[input:" + (input instanceof FileInput ? ((FileInput) input).getFileInfo() : input.toString()) + "]";</div>
<div>65&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    }</div>
<div>66&emsp;&emsp;</div>
<div>67&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    /**</div>
<div>68&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * Returns the input source that was causing this exception, for instance an</div>
<div>69&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * input file that was passed to a command. Returns null if it is unknown.</div>
<div>70&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     *</div>
<div>71&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @return the input source or file that was causing the exception, or null</div>
<div>72&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * if it is not known</div>
<div>73&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     */</div>
<div>74&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    public Input getInput() {</div>
<div>75&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        return input;</div>
<div>76&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    }</div>
<div>77&emsp;&emsp;</div>
<div>78&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    /**</div>
<div>79&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * Sets the input source that was causing this exception, for instance an</div>
<div>80&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * input file that was passed to a command.</div>
<div>81&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     *</div>
<div>82&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     * @param input the input source or file that was causing the exception</div>
<div>83&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;     */</div>
<div>84&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    public void setInput(Input input) {</div>
<div>85&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;        this.input = input;</div>
<div>86&emsp;&emsp;&nbsp;&nbsp;&nbsp;&nbsp;    }</div>
<div>87&emsp;&emsp;}</div>
</div>
</div>
</body>
</html>